### ISSUE TITLE
`java.lang.StringBuffer` cannot be deserialized

### ISSUE THREAD

**Issue by Lolf1010** [FasterXML/jackson-databind#3446]
**Describe the bug**
I recently tried to update my jackson-version from 2.10.3 to 2.13.1 and it appears that the java.lang.StringBuffer can no longer be deserialized. java.lang.StringBuilder and java.lang.String both work as expected.

Is there a reason why the support for StringBuffer was removed somewhere between the versions or is it a bug?

**Version information**
2.13.1

**To Reproduce**
```java
ObjectMapper mapper = new ObjectMapper();
StringBuffer a = new StringBuffer("test");
String json = mapper.writeValueAsString(a);                  // is the string "test"
StringBuffer b = mapper.readValue(json, StringBuffer.class); // throws Exception
```

```java
com.fasterxml.jackson.databind.exc.MismatchedInputException: Cannot construct instance of `java.lang.StringBuffer` (although at least one Creator exists): no String-argument constructor/factory method to deserialize from String value ('test')
```

**Expected behavior**
java.lang.String, java.lang.StringBuilder, java.lang.StringBuffer can be serialized and deserialized.

**Comment by GedMarc** [comment]
Uhhhh, StringBuffer, is not recommended for use at all, why are you using it?

**Comment by Lolf1010** [comment]
I am not using it directly.
Gradle is using it.
https://docs.groovy-lang.org/latest/html/groovy-jdk/java/lang/String.html#leftShift(java.lang.Object)

At the project i am working at, where this issue occurred, is it possible for developers to define and execute scripts in gradle as part of bpmn-models that can be executed. Instead of using "+" to append strings, some where using "<<" which creates a StringBuffer instead of a String - for whatever reason.

**Comment by cowtowncoder** [comment]
@Lolf1010 I can't remember any explicit change to cause this, so does sound like a bug.

At practical level, if there is no unit test to test certain functionality, it is not really supported feature -- if there was, we'd have had regression failure for whatever change occurred.
So I suspect only `String` and `StringBuilder` have test coverage.

I hope to have time to check this out, probably simple to fix.

**Comment by cowtowncoder** [comment]
Forgot to add a more meaningful update here along with the fix that went in Jackson 2.13.3 version.

So. Regression occurred in 2.12.0: formerly deserialization was working "accidentally" by Bean introspection finding single-String-arg constructor and using that. Introspection of core JDK classes was limited in 2.12 for security (primary) and performance (secondary) reasons, preventing this mechanism from working. And since there were no tests verifying deserializability (I added those with the fix), regression was not noticed.

Versions prior to 2.12.0 (2.11.x, 2.10.x etc) have working deserialization, as now does 2.13(.3). `StringBuffer` as a type is now supported wrt deserialization.
